conn.go

```diff
@@ -222,7 +222,7 @@ func (p *conn) handshake(props []interface{}) error {
 	p.props[PropLocalAddr] = p.c.LocalAddr()
 	p.props[PropRemoteAddr] = p.c.RemoteAddr()
 
-	for len(props) > 2 {
+	for len(props) >= 2 {
```

Comment:
Would you like me to separate this out into another PR? Or if you want to just push a commit with it, I can rebase my branch.